tests/run-pass/sync.rs

```diff
@@ -1,18 +1,116 @@
-// Just instantiate some data structures to make sure we got all their foreign items covered.
-// Requires full MIR on Windows.
+#![feature(rustc_private)]
 
-use std::sync;
+use std::sync::{Mutex, RwLock, TryLockError};
+
+extern crate libc;
 
 fn main() {
-    let m = sync::Mutex::new(0);
-    drop(m.lock());
+    test_mutex();
+    #[cfg(not(target_os = "windows"))] // TODO: implement RwLock on Windows
+    {
+        test_mutex_libc_recursive();
+        test_rwlock_stdlib();
+        test_rwlock_libc_init();
+        test_rwlock_libc_static_initializer();
+    }
+}
+
+fn test_mutex() {
+    let m = Mutex::new(0);
+    {
+        let _guard = m.lock();
+        assert!(m.try_lock().unwrap_err().would_block());
+    }
+    drop(m.try_lock().unwrap());
```

Comment:
Looks like this line fails on Windows.

Curiously, the `assert!(m.try_lock().unwrap_err().would_block())` succeeds.

Reply:
The reason this fails is that the `unwrap` fails, then we get a panic, and panics don't work on Windows. Probably the Windows `try_lock` stub just always returns failure...

Reply:
In https://github.com/rust-lang/miri/pull/1196 I adjusted the Windows situation to at least not be incorrect any more.

It is probably best to just entirely disable this test on Windows... the basic acquire/release functionality is tested by the libstd runtime (before-main), the rest doesn't work anyway.

Reply:
Or... hm. Looking at https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-tryentercriticalsection, I think it would be correct for us to just make `TryEnterCriticalSection` always return something non-zero. Then the `Mutex` test you wrote should pass.

I'll leave it up to you if you want to pursue this.